CMakeLists.txt

```diff
@@ -175,8 +175,6 @@ option(WITH_ELASTICSEARCH
 
 option(WITH_ZPAGES "Whether to include the Zpages Server in the SDK" OFF)
 
-option(WITH_JAEGER "DEPRECATED - Whether to include the Jaeger exporter" OFF)
```

Comment:
Could we keep the option but only allow the default value? I mean if we delete the option, the existing cmake invocation like `cmake -DWITH_JAEGER=OFF` will have warning triggered. The same applies to `cmake -DWITH_JAEGER=ON`. Better to show an explicit error message the user tries to enable Jaeger?

Reply:
> Could we keep the option but only allow the default value? I mean if we delete the option, the existing cmake invocation like `cmake -DWITH_JAEGER=OFF` will have warning triggered. The same applies to `cmake -DWITH_JAEGER=ON`. Better to show an explicit error message the user tries to enable Jaeger?

Thanks for the early comments.

The initial deprecation was announced with release 1.8.2 published on 2023-01-31.
It was advertised at great length, including with pinned issues, so users should be well aware of this by now.

With this removal, the following commands:

```
cmake -DWITH_JAEGER=ON
cmake -DWITH_JAEGER=OFF
```

will both produce a warning, with CMake indicating that an unknown option is used:

```
CMake Warning:
  Manually-specified variables were not used by the project:

    WITH_JAEGER
```

Using `-DWITH_JAEGER=OFF` produces a warning only, and will not cause the build to fail.

The fact that the `WITH_JAEGER` option is no longer available is now documented in the CHANGELOG, in the important changes section, for the next release.

I don't think the CMakeList.txt should still contain logic about obsolete options, as it makes _removing_ options impossible, this is mitigated by the important changes in the CHANGELOG instead.